client/lib/media/store.js

```diff
@@ -31,6 +31,15 @@ function receiveSingle( siteId, item, itemId ) {
 		}
 
 		MediaStore._pointers[ siteId ][ itemId ] = item.ID;
+
+		const maybeTransientMediaItem = MediaStore._media[ siteId ][ itemId ];
+
+		if ( maybeTransientMediaItem && maybeTransientMediaItem.transient ) {
```

Comment:
We can use `isItemBeingUploaded` from media utils for this check. It makes it a bit clearer as to what's happening too.